### PR TITLE
Added param conversion for notch filter 

### PR DESCRIPTION
**Pull request by tridge**
This extracts just the param conversion code from #20521 
this is a lower risk change for 4.2.0, then we can put the full 2nd harmonic notch into 4.2.1

### CODE REVIEW

File: ArduCopter/Parameters.cpp [hunk 1]
```diff
@@ -1387,14 +1387,22 @@ void Copter::convert_pid_parameters(void)
...
 
-    // notch filter parameter conversions (changed position and type) for Copter-3.7
-    const AP_Param::ConversionInfo notchfilt_conversion_info[] = {
-        { Parameters::k_param_ins, 165, AP_PARAM_INT16, "INS_NOTCH_FREQ" },
-        { Parameters::k_param_ins, 229, AP_PARAM_INT16, "INS_NOTCH_BW" },
-    };
-    uint8_t notchfilt_table_size = ARRAY_SIZE(notchfilt_conversion_info);
-    for (uint8_t i=0; i<notchfilt_table_size; i++) {
-        AP_Param::convert_old_parameters(&notchfilt_conversion_info[i], 1.0f);
+    if (!ins.gyro_notch_enabled()) {
```
Comment (andyp1per) [inline ArduCopter/Parameters.cpp:1390]: shouldn't this be ```if (ins.gyro_notch_enabled())```?

File: ArduCopter/Parameters.cpp [hunk 2]
```diff
@@ -1387,14 +1387,22 @@ void Copter::convert_pid_parameters(void)
...
+            { Parameters::k_param_ins, 101, AP_PARAM_INT8,  "INS_HNTC2_ENABLE" },
+            { Parameters::k_param_ins, 293, AP_PARAM_FLOAT, "INS_HNTC2_ATT" },
+            { Parameters::k_param_ins, 357, AP_PARAM_FLOAT, "INS_HNTC2_FREQ" },
+            { Parameters::k_param_ins, 421, AP_PARAM_FLOAT, "INS_HNTC2_BW" },
+        };
+        uint8_t notchfilt_table_size = ARRAY_SIZE(notchfilt_conversion_info);
+        for (uint8_t i=0; i<notchfilt_table_size; i++) {
+            AP_Param::convert_old_parameters(&notchfilt_conversion_info[i], 1.0f);
+        }
+        if (ins.gyro_notch_enabled()) {
```
Comment (andyp1per) [inline ArduCopter/Parameters.cpp:1402]: this will never trigger because of the if above

File: ArduPlane/Parameters.cpp [hunk 3]
```diff
@@ -1502,5 +1502,23 @@ void Plane::load_parameters(void)
     }
 #endif
 
+    if (!ins.gyro_notch_enabled()) {
```
Comment (andyp1per) [inline ArduPlane/Parameters.cpp:1505]: ditto

File: ArduPlane/Parameters.cpp [hunk 4]
```diff
@@ -1502,5 +1502,23 @@ void Plane::load_parameters(void)
...
+            { Parameters::k_param_ins, 101, AP_PARAM_INT8,  "INS_HNTC2_ENABLE" },
+            { Parameters::k_param_ins, 293, AP_PARAM_FLOAT, "INS_HNTC2_ATT" },
+            { Parameters::k_param_ins, 357, AP_PARAM_FLOAT, "INS_HNTC2_FREQ" },
+            { Parameters::k_param_ins, 421, AP_PARAM_FLOAT, "INS_HNTC2_BW" },
+        };
+        uint8_t notchfilt_table_size = ARRAY_SIZE(notchfilt_conversion_info);
+        for (uint8_t i=0; i<notchfilt_table_size; i++) {
+            AP_Param::convert_old_parameters(&notchfilt_conversion_info[i], 1.0f);
+        }
+        if (ins.gyro_notch_enabled()) {
```
Comment (andyp1per) [inline ArduPlane/Parameters.cpp:1517]: ditto

File: ArduCopter/Parameters.cpp (hunk 1 above)
Comment (tridge) [reply ArduCopter/Parameters.cpp:1390]: no, it should only convert if the 2nd notch filter isn't enabled. If it is enabled then the user has set some notch parameters

File: ArduPlane/Parameters.cpp (hunk 3 above)
Comment (tridge) [reply ArduPlane/Parameters.cpp:1505]: double ditto

File: ArduPlane/Parameters.cpp (hunk 4 above)
Comment (tridge) [reply ArduPlane/Parameters.cpp:1517]: triple ditto :-)

File: ArduCopter/Parameters.cpp (hunk 2 above)
Comment (tridge) [reply ArduCopter/Parameters.cpp:1402]: yes, it will trigger, if the conversion led to it becoming enabled

**Review by andyp1per — APPROVED** [review]

**Pull request merged**